pkg/gateway/errors/errors.go

```diff
@@ -758,4 +759,9 @@ var Codes = errorCodeMap{
 		Description:    "Attempted to write to a protected branch",
 		HTTPStatusCode: http.StatusForbidden,
 	},
+	ErrReadOnlyRepository: {
+		Code:           "ErrReadOnlyRepository",
+		Description:    "Attempted to write to a read-only repository",
+		HTTPStatusCode: http.StatusForbidden,
+	},
```

Comment:
Please return it in update operations (e.g [delete object](https://github.com/treeverse/lakeFS/blob/1b60f9d7fffdec4fb9b7f25a9b7c32c3c0fdbe38/pkg/gateway/operations/deleteobject.go#L78))